src/components/calendar/calendar.jsx

```diff
@@ -208,6 +248,12 @@ function Calendar(props) {
     );
   };
 
+  const onYearKeyDown = (event) => {
```

Comment:
Instead of adding a handle on each list option, you might be able to use the Listbox interface for `onChange`. Instead of re-inventing a11y support on each ListOption, we can just react to the listbox getting a new value.